x86: enable kmod-tg3 on 64-bit by default

bobafetthotmail opened the pull request:
Gigabit ethernet adapters using BCM5719/5720 chipset
are common on servers and as easy/cheap to get as
Intel based ones.
Usually found in 2-port and 4-port cards.

Also some devices recently added to x86_64 target
like the Meraki MX100 use this chipset for 8 of
their 12 integrated ports.

Signed-off-by: Alberto Bursi <alberto.bursi@outlook.it>

This is basically a followup of https://github.com/openwrt/openwrt/commit/9c4f903999d8303eba5e1972c6afe5ae7e2a9d39 where I added  kmod-bnx2 for older gen Broadcomm gbit eth chipsets

bobafetthotmail commented:
hot damn, on Github I stll have the old email and it's signing it wrong. Will fix this in a bit

bobafetthotmail commented:
fixed the signed-off-by

chunkeey commented:
I think github still ate it?! 

This is from the raw .patch file

https://github.com/openwrt/openwrt/commit/0989f640f643cbba152d01cdc5c7aa255e58f200.patch

    From: Alberto Bursi <bobafetthotmail@gmail.com>
    Date: Tue, 2 Nov 2021 12:34:04 +0100
    Subject: [PATCH] x86: enable kmod-tg3 on 64-bit by default

    Gigabit ethernet adapters using BCM5719/5720 chipset
    [...]
    Signed-off-by: Alberto Bursi <alberto.bursi@outlook.it>

The ````From:```` and ````Signed-off-By: ```` have to match, otherwise the git hooks will complain.

aparcar commented:
Would you be okay to walk the extra mile and split them by line and sort them alphabetically?

bobafetthotmail commented:
> The `From:` and `Signed-off-By: ` have to match, otherwise the git hooks will complain.

yeah I know, that's the problem I had. I did a **git commit --amend** from my PC where git has the correct email but it seems this still isn't enough (although now the DCO test is passing, it was not passing before)

seems like I have to delete the commit and create it again

> Would you be okay to walk the extra mile and split them by line and sort them alphabetically?

yeah sure, will do a separate commit for that. What do you mean with "split by line" exactly?

bobafetthotmail committed:
x86: enable kmod-tg3 on 64-bit by default

Gigabit ethernet adapters using BCM5719/5720 chipset
are common on servers and as easy/cheap to get as
Intel based ones.
Usually found in 2-port and 4-port cards.

Also some devices recently added to x86_64 target
like the Meraki MX100 use this chipset for 8 of
their 12 integrated ports.

Signed-off-by: Alberto Bursi <alberto.bursi@outlook.it>

bobafetthotmail committed:
x86: sort alphabetically default packages

sorting alphabetically default packages
and placing them on their own line.

Signed-off-by: Alberto Bursi <alberto.bursi@outlook.it>

bobafetthotmail commented:
done and now looks good in the raw too https://github.com/openwrt/openwrt/commit/7c1ed675a77f6bc5dc0d933364f0512c314f2e17.patch

Added a commit after it to reorder alphabetically the packages

aparcar commented:
Oh I meant to split it into multiple lines, one per package. But this is fine too

chunkeey commented:
Merged. Thanks!

Hurricos commented:
Oh my goodness. I needed this to get MX100 port enumeration to stabilize and remembered [@bobafetthotmail mentioning what I should go do](https://forum.openwrt.org/t/broadcom-bcm5719-pci-e-supported/58437/4). I'm very appreciative to see this already done.

Now to wait for Snapshot to build ...
